src/Microsoft.Health.SqlServer.UnitTests/ManagedIdentitySqlConnectionTests.cs

```diff
@@ -18,47 +21,62 @@ public class ManagedIdentitySqlConnectionTests
 
-    public ManagedIdentitySqlConnectionTests()
-    {
-        var accessTokenHandler = Substitute.For<IAccessTokenHandler>();
-        accessTokenHandler.GetAccessTokenAsync(AzureResource).Returns(Task.FromResult(TestAccessToken));
-
-        SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration = new SqlServerDataStoreConfiguration
-        {
-            ConnectionString = $"Server={ServerName};Database={DatabaseName};",
-            AuthenticationType = SqlServerAuthenticationType.ManagedIdentity,
-        };
 
-        var sqlConfigOptions = Options.Create(sqlServerDataStoreConfiguration);
-        _sqlConnectionFactory = new ManagedIdentitySqlConnectionBuilder(
-            new DefaultSqlConnectionStringProvider(sqlConfigOptions), accessTokenHandler, SqlConfigurableRetryFactory.CreateNoneRetryProvider());
-    }
 
-    [Fact]
-    public async Task GivenManagedIdentityConnectionType_WhenSqlConnectionRequested_AccessTokenIsSet()
+    [Theory]
+    [InlineData(SqlServerAuthenticationType.ManagedIdentity)]
+    [InlineData(SqlServerAuthenticationType.WorkloadIdentity)]
+    public async Task GivenManagedIdentityConnectionType_WhenSqlConnectionRequested_AccessTokenIsSet(SqlServerAuthenticationType sqlServerAuthenticationType)
     {
+        InitializeTest(sqlServerAuthenticationType);
+
         SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync().ConfigureAwait(false);
 
         Assert.Equal(TestAccessToken, sqlConnection.AccessToken);
     }
 
-    [Fact]
-    public async Task GivenDefaultConnectionType_WhenSqlConnectionRequested_DatabaseIsSet()
+    [Theory]
+    [InlineData(SqlServerAuthenticationType.ManagedIdentity)]
+    [InlineData(SqlServerAuthenticationType.WorkloadIdentity)]
+    public async Task GivenDefaultConnectionType_WhenSqlConnectionRequested_DatabaseIsSet(SqlServerAuthenticationType sqlServerAuthenticationType)
     {
+        InitializeTest(sqlServerAuthenticationType);
         SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync().ConfigureAwait(false);
 
         Assert.Equal(DatabaseName, sqlConnection.Database);
     }
 
-    [Fact]
-    public async Task GivenDefaultConnectionType_WhenSqlConnectionToMasterRequested_MasterDatabaseIsSet()
+    [Theory]
+    [InlineData(SqlServerAuthenticationType.ManagedIdentity)]
+    [InlineData(SqlServerAuthenticationType.WorkloadIdentity)]
+    public async Task GivenDefaultConnectionType_WhenSqlConnectionToMasterRequested_MasterDatabaseIsSet(SqlServerAuthenticationType sqlServerAuthenticationType)
     {
+        InitializeTest(sqlServerAuthenticationType);
         SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync(MasterDatabase).ConfigureAwait(false);
 
         Assert.Equal(MasterDatabase, sqlConnection.Database);
     }
+
+    private void InitializeTest(SqlServerAuthenticationType sqlServerAuthenticationType)
```

Comment:
You can presumably move this back, right? As `ManagedIdentitySqlConnectionBuilder` doesn't actually read the `AuthenticationType` property

Reply:
Then we can move it back to the ctor